http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandlerSpec.scala

```diff
@@ -375,19 +379,184 @@ class AzurePubsubHandlerSpec
+
+    val failAzureVmService = new FakeAzureVmService {
+      override def startAzureVm(name: InstanceName, cloudContext: AzureCloudContext)(implicit
+        ev: Ask[IO, TraceId]
+      ): IO[Option[Mono[Void]]] = IO.none
+    }
+
+    val queue = QueueFactory.asyncTaskQueue()
+
+    val azureInterp =
+      makeAzureInterp(asyncTaskQueue = queue, azureVmService = failAzureVmService)
+
+    val res = for {
+      disk <- makePersistentDisk().copy(status = DiskStatus.Ready).save()
+      azureRuntimeConfig = RuntimeConfig.AzureConfig(MachineTypeName(VirtualMachineSizeTypes.STANDARD_A1.toString),
+                                                     disk.id,
+                                                     azureRegion
+      )
+      runtime = makeCluster(1)
+        .copy(
+          cloudContext = CloudContext.Azure(azureCloudContext)
+        )
+        .saveWithRuntimeConfig(azureRuntimeConfig)
+
+      _ <- azureInterp.startAndMonitorRuntime(runtime, azureCloudContext)
+
+    } yield ()
+    a[RuntimeNotFoundException] should be thrownBy {
+      res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+    }
+  }
+
+  it should "stop azure vm" in isolatedDbTest {
+    // Set up virtual machine mock.
+    val vmReturn = mock[VirtualMachine]
+    when(vmReturn.powerState()).thenReturn(PowerState.RUNNING)
+
+    val passAzureVmService = new FakeAzureVmService {
+      override def stopAzureVm(name: InstanceName, cloudContext: AzureCloudContext)(implicit
+        ev: Ask[IO, TraceId]
+      ): IO[Option[Mono[Void]]] = IO.some(Mono.empty[Void]())
+    }
+    val queue = QueueFactory.asyncTaskQueue()
+
+    val azureInterp =
+      makeAzureInterp(asyncTaskQueue = queue, azureVmService = passAzureVmService)
+
+    val res = for {
+      disk <- makePersistentDisk().copy(status = DiskStatus.Ready).save()
+      azureRuntimeConfig = RuntimeConfig.AzureConfig(MachineTypeName(VirtualMachineSizeTypes.STANDARD_A1.toString),
+                                                     disk.id,
+                                                     azureRegion
+      )
+      runtime = makeCluster(1)
+        .copy(
+          cloudContext = CloudContext.Azure(azureCloudContext)
+        )
+        .saveWithRuntimeConfig(azureRuntimeConfig)
+
+      // Assertions for runtime
+      assertions = for {
+        getRuntimeOpt <- clusterQuery.getClusterById(runtime.id).transaction
+        getRuntime = getRuntimeOpt.get
+      } yield getRuntime.status shouldBe RuntimeStatus.Stopped
+
+      // Mocked response, should succeed into the Some(mono) case
+      asyncTaskProcessor = AsyncTaskProcessor(AsyncTaskProcessor.Config(10, 10), queue)
+      _ <- azureInterp.stopAndMonitorRuntime(runtime, azureCloudContext)
+
+      _ <- withInfiniteStream(asyncTaskProcessor.process, assertions)
+
+    } yield ()
+
+    res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
+  }
+
+  it should "fail on stopAzureVm - no runtime found" in isolatedDbTest {
```

Comment:
it's more valuable to test that we're handling a `AzureRuntimeStopError` properly than to verify an arbitrary exception is thrown